solutions/silver/usaco-895.mdx

```diff
@@ -19,74 +26,87 @@ import java.io.*;
 import java.util.*;
 
 public class IcyPerimeter {
-	public static void main(String[] args) throws IOException {
-		Kattio io = new Kattio("perimeter");
-		n = io.nextInt();
-
-		// we do n + 2 because there is an extra outer ring for flood fill
-		numberedMatrix = new int[n + 2][n + 2];
-
-		for (int x = 1; x <= n; x++) {  // read matrix
-			String line = io.next();
-			for (int y = 1; y <= n; y++) {
-				String character = Character.toString(line.charAt(y - 1));
-				if (character.equals("#")) {
-					numberedMatrix[x][y] = -1;  // mark # as -1
+	// Movement directions (up, down, left, right)
+	private static final int[][] DIRECTIONS = {{-1, 0}, {1, 0}, {0, -1}, {0, 1}};
+
+	private static int[] bfs(int row, int col, char[][] grid, boolean[][] visited,
+	                         int n) {
+		Deque<int[]> q = new ArrayDeque<>();
+		q.add(new int[] {row, col});  // Start BFS from the given cell (row, col)
+
+		int area = 0;
+		int perimeter = 0;
+
+		while (!q.isEmpty()) {
+			int[] current = q.poll();  // Current cell (row, col)
+			row = current[0];
+			col = current[1];
+
+			// Skip if the cell has already been visited
+			if (visited[row][col]) { continue; }
+
+			visited[row][col] = true;  // Mark the cell (row, col) as visited
+			area++;  // Increment area (each '#' contributes 1 to the area)
+
+			int sides =
+			    4;  // Each cell starts with 4 sides contributing to the perimeter
```

Comment:
hmm precommit is tweaking out, can u move this comment to a be above the variable declaration?